### PR TITLE
Userprofile

### PR DESCRIPTION
**Pull request by regilkent123**

### CODE REVIEW

File: OJT/settings.py [hunk 1]
```diff
@@ -105,7 +107,7 @@
 
 LANGUAGE_CODE = 'en-us'
 
-TIME_ZONE = 'UTC'
+TIME_ZONE = 'Hongkong'
```
Comment (ozknightwalker) [inline OJT/settings.py]: @regilkent123 don't change this one. its not needed

File: OJT/views.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from django.shortcuts import render_to_response
+from django.http import HttpResponseRedirect
+from django.contrib import auth
+from django.core.context_processors import csrf
+
+def login(request):
```
Comment (ozknightwalker) [inline OJT/views.py]: @regilkent123 @iamkatrinaanne please use classbased views. [documentation](https://docs.djangoproject.com/en/2.0/topics/class-based-views/)

File: OJT/views.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    c= {}
+    c.update(csrf(request))
+    return render_to_request('login.html',c)
+def auth_view(request):
```
Comment (ozknightwalker) [inline OJT/views.py]: @regilkent123 @iamkatrinaanne please use classbased views. [documentation](https://docs.djangoproject.com/en/2.0/topics/class-based-views/)

File: OJT/views.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    password=request.POST.get('password','')
+    user=auth.authenticate(username=username, password=password)
+
+    if user is not None:
+        auth.login(request,user)
+        return HttpResponseRedirect('login/')
+    else :
+        return HttpResponseRedirect('invalidlogin/')
+
+def loggedin(request)
```
Comment (ozknightwalker) [inline OJT/views.py]: @regilkent123 @iamkatrinaanne please use classbased views. [documentation](https://docs.djangoproject.com/en/2.0/topics/class-based-views/)

File: OJT/views.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    return render_to_response('loggedin.html',{'full_name' request.user.username})
+
+def invalid_login(request)
```
Comment (ozknightwalker) [inline OJT/views.py]: @regilkent123 @iamkatrinaanne please use classbased views. [documentation](https://docs.djangoproject.com/en/2.0/topics/class-based-views/)

File: OJT/views.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    return to render_to_response('invalid_login.html')
+
+def logout(request)
```
Comment (ozknightwalker) [inline OJT/views.py]: @regilkent123 @iamkatrinaanne please use classbased views. [documentation](https://docs.djangoproject.com/en/2.0/topics/class-based-views/)

File: userprofile/admin.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (ozknightwalker) [inline userprofile/admin.py:3]: @regilkent123 @iamkatrinaanne please register the profile models here so you can see the profile model via admin page

File: userprofile/templates/userprofile/login.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
+
+{% extends 'base.html' %}
+{% block content %}
+
+<link rel="stylesheet" type="text/css" href="../static/userprofile/styles.css" />
```
Comment (ozknightwalker) [inline userprofile/templates/userprofile/login.html]: @iamkatrinaanne @regilkent123 https://docs.djangoproject.com/en/2.0/howto/static-files/

File: userprofile/templates/userprofile/login.html [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+
+<div class="container">
+    <form method="post" action="/loggedin/">
+        {% csrf_token %}
+        <div class="login">
+            <div id="logo"><img src="../static/userprofile/images/edit.png">
```
Comment (ozknightwalker) [inline userprofile/templates/userprofile/login.html]: @regilkent123 @iamkatrinaanne https://docs.djangoproject.com/en/2.0/howto/static-files/

File: OJT/views.py (hunk 2 above)
Comment (ozknightwalker) [reply OJT/views.py]: @regilkent123 @iamkatrinaanne wait, is this view used anywhere? if not please remove this

File: OJT/views.py (hunk 3 above)
Comment (ozknightwalker) [reply OJT/views.py]: @regilkent123 @iamkatrinaanne wait, is this view used anywhere? if not please remove this

File: OJT/views.py (hunk 4 above)
Comment (ozknightwalker) [reply OJT/views.py]: @regilkent123 @iamkatrinaanne wait, is this view used anywhere? if not please remove this

File: OJT/views.py (hunk 5 above)
Comment (ozknightwalker) [reply OJT/views.py]: @regilkent123 @iamkatrinaanne wait, is this view used anywhere? if not please remove this

File: OJT/views.py (hunk 6 above)
Comment (ozknightwalker) [reply OJT/views.py]: @regilkent123 @iamkatrinaanne wait, is this view used anywhere? if not please remove this

File: userprofile/views.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    u.set_password(password)
+    u.save()
+
+    return HttpResponse('success')
+
+class LoginView(generic.TemplateView):
+    template_name = 'userprofile/login.html'
+
+
+def auth_login(request):
```
Comment (ozknightwalker) [inline userprofile/views.py:59]: @regilkent123 @iamkatrinaanne  please use classbased views here

File: userprofile/views.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+from django.views import generic
+from django.contrib.auth import authenticate, login
+from django.urls import reverse
+from django.http import HttpResponseRedirect, HttpResponse
+from django.contrib.auth.models import User
+
+class RegisterView(generic.TemplateView):
+    template_name = 'userprofile/register.html'
+
+def register(request):
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne please use classbased views here

File: userprofile/views.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    usernames = request.POST['username']
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne make use of django forms [documentation](https://docs.djangoproject.com/en/2.0/topics/forms/)

```
form = UserForm(request.POST)
if form.is_valid():
    ...create user
else:
    ... show an error message 
```

File: OJT/views.py (hunk 5 above)
Comment (regilkent123) [reply OJT/views.py]: @ozknightwalker delete kuya hahaha nasayop ug butang

**Pull request merged**